pkg/controller/clusters/status.go

```diff
@@ -22,6 +22,11 @@ func UpdateClusterStatusForControlPlane(ctx context.Context, client client.Clien
 		return errors.Wrapf(err, "getting kubeadmcontrolplane")
 	}
 
+	if kcp == nil {
```

Comment:
why this change? it's difficult to understand the changes without any context

Reply:
it caused a nil pointer dereference when deleting a workload cluster in TestVSphereUpgradeKubernetes123to124UbuntuWorkloadClusterAPI in updateControlPlaneReadyCondition(cluster, kcp)

Reply:
I'd vote to just return nil here instead of also setting the condition then. The ControlPlaneInitialized condition should not be updated again after being "True". This will update the value back to "False"

Reply:
is it easier to just add this check to `updateControlPlaneReadyCondition`?

Reply:
yeah we can do that instead

Reply:
yeah, i think that solution is better